### PR TITLE
Devnet updates: Genesis with CW XCVM contracts for Substrate devnet and Osmosisd app

### PR DESCRIPTION
**Pull request by dzmitry-lahoda**
- CW in CI green and builds all the time
- Osmosisd available
- CW in genesis of Picasso (some issue with interpreter contract in genesis) 
- Centauri chain update to latest version and process-compose update
- relayer of Centauri <-> Osmosis

next:
- centauiry hack for pica denom
- CW genesis in Osmosis/Centauri, init, start
- Memo limit for Osmosis 2x increase hack
- composable node with isolation for p2p discovery and 3rd relayer
- macos darwin support (seem couple of lines + build in CI on main)
- starting from prebaked genesis

- [x] PR title is my best effort to provide summary of changes and has clear text to be part of release notes 
- [x] I marked PR by `misc` label if it should not be in release notes
- [ ] I have linked Zenhub/Github or any other reference item if one exists
- [x] I was clear on what type of deployment required to release my changes (node, runtime, contract, indexer, on chain operation, frontend, infrastructure) if any in PR title or description
- [x] I waited and did best effort for `pr-workflow-check / draft-release-check` to finish with success(green check mark) with my changes
- [x] I have added at least one reviewer in reviewers list
- [x] I tagged(@) or used other form of notification of one person who I think can handle best review of this PR
- [x] I have proved that PR has no general regressions of relevant features and processes required to release into production

### CODE REVIEW

File: code/xcvm/cosmwasm/tests/src/tests/suite.rs [hunk 1]
```diff
@@ -317,29 +317,21 @@ fn load_contracts() -> XCVMContracts {
...
+		read(contracts_dir.join(filename))
 	}
 
-	let code_interpreter = read_contract("CW_XCVM_INTERPRETER", "cw_xc_interpreter.wasm");
-	let code_gateway = read_contract("CW_XCVM_GATEWAY", "cw_xc_gateway.wasm");
-
+	let code_interpreter = read_contract("cw_xc_interpreter.wasm");
+	let code_gateway = read_contract("cw_xc_gateway.wasm");
+	let out_dir =
+		option_env!("NIX_CARGO_OUT_DIR").unwrap_or(env!("OUT_DIR")).to_owned() + "/cw20_base.wasm";
```
Comment (mina86) [inline code/xcvm/cosmwasm/tests/src/tests/suite.rs:331]: Alternatively:
```suggestion
		[option_env!("NIX_CARGO_OUT_DIR").unwrap_or(env!("OUT_DIR")), "/cw20_base.wasm"].concat();
```
It’s slightly longer but forgoes reallocation when concatenating.

File: code/xcvm/cosmwasm/tests/src/tests/suite.rs [hunk 2]
```diff
@@ -317,29 +317,21 @@ fn load_contracts() -> XCVMContracts {
...
-		}
+		read(contracts_dir.join(filename))
 	}
 
-	let code_interpreter = read_contract("CW_XCVM_INTERPRETER", "cw_xc_interpreter.wasm");
-	let code_gateway = read_contract("CW_XCVM_GATEWAY", "cw_xc_gateway.wasm");
-
+	let code_interpreter = read_contract("cw_xc_interpreter.wasm");
+	let code_gateway = read_contract("cw_xc_gateway.wasm");
+	let out_dir =
```
Comment (mina86) [inline code/xcvm/cosmwasm/tests/src/tests/suite.rs:330]:
```suggestion
	let cw20_base_path =
```
or some such.  The variable doesn’t hold output directory path.

File: code/parachain/node/src/chain_spec/picasso.rs [hunk 3]
```diff
@@ -25,6 +25,25 @@ pub fn genesis_config(
...
+		// /build/source/parachain/frame/cosmwasm/src/lib.rs:406 contracts.push(contract);
+	}
+
+	let cosmwasm = picasso_runtime::CosmwasmConfig {
+		contracts: contracts
+			.into_iter()
+			.map(|x| (root.clone(), x.try_into().expect("genesis cw wasm fits constraints")))
+			.collect(),
+		..Default::default()
+	};
```
Comment (mina86) [inline code/parachain/node/src/chain_spec/picasso.rs:45]:
```suggestion
	if let Some(contract) = option_env!("CW_XC_GATEWAY_WASM_PATH") {
		contracts.push(contract);
	}
	if let Some(contract) = include_contract(option_env!("CW_XC_INTERPRETER_WASM_PATH")) {
		// not sure what is going on, but it is weird
		// Thread 'main' panicked at 'contracts in genesis are valid: Module(ModuleError { index:
		// 180, error: [4, 0, 0, 0], message: Some("CodeValidation") })',
		// /build/source/parachain/frame/cosmwasm/src/lib.rs:406 contracts.push(contract);
	}

	let contracts = contracts
		.into_iter()
		.map(|path| match std::fs::read(path).map(|bytes| bytes.try_into()) {
			Ok(Ok(data)) => contract,
			Ok(Err(err)) => panic!("{path}: {err}"),
			Err(err) => panic!("{path}: {err}"),
		})
		.map(|contract| (root.clone(), contract))
		.collect();

	let cosmwasm = picasso_runtime::CosmwasmConfig {
		contracts,
		..Default::default(),
	};
```

and then drop `include_contract` function.  There’s no need to open
and read the file by using `Read` trait there.  `std::fs::read` does
that for you.

**Review by mina86 — APPROVED** [review]
Rust parts LGTM.  The other parts seem sensible to me too.

**Review by RustNinja — DISMISSED** [review]

File: code/parachain/node/src/chain_spec/picasso.rs [hunk 4]
```diff
@@ -26,7 +26,7 @@ pub fn genesis_config(
 	treasury: AccountId,
 ) -> picasso_runtime::GenesisConfig {
 	let mut contracts = Vec::new();
-	if let Some(contract) = include_contract(option_env!("CW_XC_GATEWAY_WASM_PATH")) {
+	if let Some(contract) = option_env!("CW_XC_GATEWAY_WASM_PATH") {
 		contracts.push(contract);
 	}
 	if let Some(contract) = include_contract(option_env!("CW_XC_INTERPRETER_WASM_PATH")) {
```
Comment (mina86) [inline code/parachain/node/src/chain_spec/picasso.rs:32]: Sorry, missed this one.  This should also drop `include_contract` call and than you can drop the function altogether.

```suggestion
	if let Some(contract) = option_env!("CW_XC_INTERPRETER_WASM_PATH") {
```

**Review by RustNinja — APPROVED** [review]

File: code/parachain/node/src/chain_spec/picasso.rs [hunk 5]
```diff
@@ -93,3 +118,14 @@ pub fn genesis_config(
 		release_committee: Default::default(),
 	}
 }
+
+fn include_contract(wasm_path: Option<&str>) -> Option<Vec<u8>> {
```
Comment (kkast) [inline code/parachain/node/src/chain_spec/picasso.rs:122]: is this used anywhere?

Comment (dzmitry-lahoda) [reply code/parachain/node/src/chain_spec/picasso.rs:122]: removed

**Pull request merged**